src/symbolize/gimli/mmap_fake.rs

```diff
@@ -0,0 +1,24 @@
+use super::{mystd::io::Read, File};
+use alloc::vec::Vec;
+use core::ops::Deref;
+
+pub struct Mmap {
+    vec: Vec<u8>,
+}
+
+impl Mmap {
+    pub unsafe fn map(mut file: &File, len: usize) -> Option<Mmap> {
+        let mut mmap = Mmap { vec: Vec::new() };
```

Comment:
Perhaps this could do `with_capacity` to preallocate? Also I think it's fine to avoid the `truncate` since this is only ever used for reading the whole file anyway.

Reply:
I've done the latter. I was having issues with `map` getting passed a length *much* larger than the file during testing, so I didn't use a `with_capacity`. It didn't affect anything else, so I didn't debug much further, but I could try to do some more debugging on that.

Reply:
Yes I think it would be good to debug that because this just calls `fs::metadata` effectively before calling this function, and the file shouldn't change that much in the meantime.

Reply:
I can't reproduce this anymore, so...